src/libarena/lib.rs

```diff
@@ -211,6 +216,7 @@ impl<T> TypedArena<T> {
     #[cold]
     fn grow(&self, n: usize) {
         unsafe {
+            let elem_size = cmp::max(1, mem::size_of::<T>());
```

Comment:
leave a comment here that we need the element size in order to go from byte sizes of `HUGE_PAGE` and `PAGE` to element counts obtained from `capacity` and used for creating new typed chunks.